java/java.project/src/org/netbeans/spi/java/project/support/ProjectPlatform.java

```diff
@@ -76,28 +75,53 @@ public static JavaPlatform forProject(
             @NonNull final String platformType) {
         Parameters.notNull("owner", owner); //NOI18N
         Parameters.notNull("eval", eval); //NOI18N
-        Parameters.notNull("platformType", platformType);   //NOI18N        
-        final String platformName = eval.getProperty(PLATFORM_ACTIVE);        
+        Parameters.notNull("platformType", platformType);   //NOI18N
+        final String platformName = eval.getProperty(PLATFORM_ACTIVE);
         final FileObject jdkHome = resolvePlatformHome(
                 platformName,
                 owner.getProjectDirectory(),
                 eval);
         if (jdkHome == null) {
             return null;
         }
+        return forProject(owner, jdkHome, platformName, platformType);
+    }
+
+    /**
+     * Creates a transient {@link JavaPlatform} defined in given {@link Project}.
+     * @param owner the project to return the {@link JavaPlatform} for
+     * @param platformHome the {@link JavaPlatform} install folder
+     * @param platformName the {@link JavaPlatform} system name uniquely identifying the platform.
+     * The name has to be valid property name, use the {@link PropertyUtils#getUsablePropertyName(java.lang.String)} to create the name.
+     * For the Ant based project the platform name is a value of {@code platform.active} property.
```

Comment:
This sentence is unlikely correct at this place.

Reply:
I will add apichange.xml.
Regarding the ` For the Ant based project...` I agree, but I have no other place to describe the contract required by Ant project. Ant project's broken project support depends on it. So I prefer to keep it here.